app/controllers/jobs_controller.rb

```diff
@@ -1,16 +1,27 @@
 class JobsController < ApplicationController
+  before_action :set_job, only: [:show, :edit, :update, :destroy]
+  before_action :set_company, except: [:all_jobs]
+  before_action :set_category_titles, only: [:new, :edit]
+
+  def all_jobs
+    @jobs = Job.all
+    if params[:sort] == 'interest'
+      @jobs = Job.order('level_of_interest DESC')
```

Comment:
this is a lot of logic in the view. i suggest encapsulating the `order(level_of_interest DESC)` in a method on your `Job`